From 7462d61442f606cc118722e0572ac8db48fa2540 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Thomas=20B=C3=A4chler?= <thomas@archlinux.org>
Date: Tue, 2 Aug 2011 10:56:33 +0200
Subject: [PATCH] rules: input - fix detection of bluetooth hid devices in
 Xorg

Commit c49df20758e0f22778cfc93b598f2929f4c86272 prevented udev from
creating broken symlinks for bluetooth hid devices. Unfortunately,
it also removed the ID_INPUT=1 and ID_INPUT_{KEY,MOUSE}=1 properties
from those devices. Xorg relies on these properties for cold- and
hotplugging of input devices.

With this patch, udev still omits the broken symlinks, but also
imports the input_id properties again.
---
 rules/rules.d/60-persistent-input.rules |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/rules/rules.d/60-persistent-input.rules b/rules/rules.d/60-persistent-input.rules
index 65a6381..28d50d2 100644
--- a/rules/rules.d/60-persistent-input.rules
+++ b/rules/rules.d/60-persistent-input.rules
@@ -2,9 +2,9 @@
 
 ACTION=="remove", GOTO="persistent_input_end"
 SUBSYSTEM!="input", GOTO="persistent_input_end"
-SUBSYSTEMS=="bluetooth", GOTO="persistent_input_end"
 
 ENV{ID_INPUT}=="", IMPORT{program}="input_id %p"
+SUBSYSTEMS=="bluetooth", GOTO="persistent_input_end"
 SUBSYSTEMS=="usb", ENV{ID_BUS}=="", IMPORT{program}="usb_id --export %p"
 
 # determine class name for persistent symlinks
-- 
1.7.6

